### PR TITLE
Fixes deprecation of `attribute_changed?` method

### PR DESCRIPTION
**Pull request by joelind**

### CODE REVIEW

File: app/models/spree/user.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class User < Spree::Base
 
     acts_as_paranoid
     after_destroy :scramble_email_and_password
-    before_update { generate_spree_api_key if encrypted_password_changed? && spree_api_key.present? }
+    before_update { generate_spree_api_key if will_save_change_to_encrypted_password? && spree_api_key.present? }
```
Comment (cbrunsdon) [inline app/models/spree/user.rb]: Where is `will_save_change_to_encrypted_password?` supposed to be defined?

Comment (tvdeyen) [reply app/models/spree/user.rb]: Looks like a new Rails 5.1 feature

https://github.com/rails/rails/pull/25337/files

We will need to check the AR version like devise does to support older versions of Solidus

**Comment by jhawthorn** [comment]
This will be fixed by #99

**Comment by tvdeyen** [comment]
Superseded by #99 

Closing. Thanks again for contributing.

**Pull request closed without merging**